src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb

```diff
@@ -4238,16 +4238,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
                     For Each expressionSyntax In node.Initializers
 
-                        Dim boundExpression As BoundExpression
-
-                        If expressionSyntax.Kind <> SyntaxKind.CollectionInitializer Then
-                            boundExpression = BindValue(expressionSyntax, diagnostics)
-
-                            If elemType IsNot Nothing Then
-                                boundExpression = ApplyImplicitConversion(expressionSyntax, elemType, boundExpression, diagnostics)
+                        Dim boundExpression As BoundExpression = BindValue(expressionSyntax, diagnostics)
```

Comment:
Should use the temp DiagnosticsBag here too, previously the expressionSyntax wasn't being bound at all.

Reply:
IMO there's no real value in reporting these nested diagnostics as they're not relevant to the issue the user is facing (too many dimensions). Technically this isn't a user-code breaking change as their code couldn't compile in the first place, but I don't see any value in reporting nested diagnostics in this scenario.

Reply:
So, I found couple of other related issues.

1. Few lines below this change, we have [another case](http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic/Binding/Binder_Expressions.vb,4270) where we report a diagnostic but don't bind the initializer if it is of unexpected syntax kind.
2. My current implementation leads to `BoundArrayLiteral` wrapped within a `BoundBadExpression` getting generated in the bound tree, but unlike other `BoundArrayLiteral` nodes which gets removed with [ReclassifyArrayLiteralExpression](http://source.roslyn.io/#Microsoft.CodeAnalysis.VisualBasic/Binding/Binder_Expressions.vb,1468), this one doesn't and this causes incorrect information from semantic model, such as the unit test failure [here](https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_debug_spanish_unit32_prtest/2240/testReport/junit/Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics/ArrayLiteralTests/TestSemanticInfoTooDeeplyNestedArrayLiteral/).

I think we need a more involved binder fix here to handle all these cases properly, and that would be risky at this point. I am going to close my PR and move the bug to 16.0.